tests/cloudpickle_test.py

```diff
@@ -1862,6 +1862,29 @@ def __getattr__(self, name):
         with pytest.raises(pickle.PicklingError, match='recursion'):
             cloudpickle.dumps(a)
 
+    def test___dict__attribute_not_dropped_during_pickling(self):
+        # Test https://github.com/cloudpipe/cloudpickle/issues/282. cloudpickle
+        # used to drop __dict__ attributes of classes at pickling time.
+        pickle_filename = os.path.join(self.tmpdir, 'class_with___dict__.pkl')
+        _dict = {'some_attribute': 1}
+        class A:
+            __dict__ = _dict
+        a = A()
+        self.assertEqual(a.__dict__, _dict)
+
+        with open(pickle_filename, "wb") as f:
+            cloudpickle.dump(a, f, protocol=self.protocol)
+
+        # Depickle the class in a new python session to make sure the class is
+        # fully-recreated, and not looked-up in existing cloudpickle
+        # class-tracking constructs.
```

Comment:
I do not understand the meaning of this comment: the class-tracking weak dicts cannot be used the first time a an instance of a dynamic class is loaded as the class definition does not exist in the Python interpreter  yet.

Reply:
This comment explains why I did not load  `depickled_a` in the interpreter session running this test: in this session, `A` is tracked, so `A`'s reconstructor would not have been called at unpickling time.

Reply:
ah ok. I misread it. Maybe you could make it more explict by rephrasing:

"Check that the dynamic class defintion is fully reconstructed in a new Python subprocess and check the assertion that the content of the __dict__ attribute has the expected content there. Testing this in the main process is not enough because the existing class definition could have been reused directly."